src/status_im2/contexts/chat/composer/handlers.cljs

```diff
@@ -133,9 +132,7 @@
     (@record-reset-fn)
     (reset! recording? false))
   (rf/dispatch [:chat.ui/set-chat-input-text text])
-  (if (string/ends-with? text "@")
-    (rf/dispatch [:mention/on-change-text text])
-    (debounce/debounce-and-dispatch [:mention/on-change-text text] 300)))
+  (rf/dispatch [:mention/on-change-text text]))
```

Comment:
Why this change? I think this is to avoid many calls to status-go for mentions cc @qfrank

Reply:
Also would like to know why 🙂

Reply:
Two things:

1) we need to get updated text structure from `status-go` as quick as possible, because while it is not updated, highlighting is not applied

2) With previous logic I experienced a bug. Imagine that user is going to type "hello @". The order of event is:
- user types "hello"
- user types whitespace after "hello", `on-change-text` scheduled with delay and with content "hello "
- user types "@", `on-change-text` immediately processed and current text input becomes "hello @"
- scheduled `on-change-text` processed with delay and current text get backs to the "hello ", which results in losing the "@"

Reply:
emm.. what if we enhance `debounce/debounce-and-dispatch` to make it support dispatch immediately if user input end with @ ?

Reply:
@qfrank, I think `debounce-and-dispatch` is too general to make this change, because it works not only with `on-change-text`, but even if we add this hack, that won't eliminate the problem with slow highlight updating.
Imagine that user typed "@asd" and continued quick typing. All the symbols after "@" won't be highlighted because while user quickly types, events not sent and updated text structure is not received from `status-go`

Reply:
yeah, quick typing would make weired behavior. 😅

> debounce-and-dispatch is too general 

what if make a new function like `debounce-and-dispatch` and add an extra trigger condition to the actual invoking e.g. the number of user input characters reached a specific value. just my thoughts , it's up to you, if we found performance issue in status-go, there still exist spaces to improve 🙂

Reply:
@qfrank, I think right now it is more a preliminary optimization) Let's see if there are any issues with existing implementation and then start improving, wdyt? I'm currently more concerned about that bug with highlighting that is in the description. I believe it can be a showstopper on slow devices :-/

Reply:
agree